src/block.rs

```diff
@@ -170,46 +170,133 @@ impl BlockBuilder {
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value2")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+        ],
+    })]
+    #[case(BlockTestCase {
+        name: "block_with_tombstone",
+        entries: vec![
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key1"),
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value1")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key2"),
+                ValueDeletable::Tombstone,
+                0,
+                Some(0),
+                Some(0),
+            ),
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key3"),
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value3")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+        ],
+    })]
+    #[case(BlockTestCase {
+        name: "block_with_merge",
+        entries: vec![
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key1"),
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value1")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key1"),
+                ValueDeletable::Merge(Bytes::copy_from_slice(b"value1")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key2"),
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value2")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+        ],
+    })]
+    fn test_block(#[case] test_case: BlockTestCase) {
+        let block = build_block(&test_case);
         let encoded = block.encode();
-        let _decoded = Block::decode(encoded);
+        let decoded = Block::decode(encoded.clone());
+        let block_data = &block.data;
+        let block_offsets = &block.offsets;
+        // Decode the block data using offsets and validate each decoded entry
+        let decoded_entries = decode_codec_entries(block_data.clone(), block_offsets)
+            .expect("Failed to decode codec entries");
+        assert_eq!(decoded_entries, test_case.entries);
+
+        assert_eq!(block_data, &decoded.data);
+        assert_eq!(block_offsets, &decoded.offsets);
+        assert_debug_snapshot!(test_case.name, (block.size(), block.data, block.offsets));
     }
 
     #[test]
-    fn test_block_size() {
-        let mut builder = BlockBuilder::new(4096);
-        assert!(builder.add_value(b"key1", b"value1", gen_attrs(1)));
-        assert!(builder.add_value(b"key2", b"value2", gen_attrs(1)));
-        let block = builder.build().unwrap();
-        assert_eq!(73, block.size());
-
+    fn test_add_tombstone() {
```

Comment:
I think it would be IMHO an overkill to do a snapshot for testing **add_tomstone**. It basically return as boolean and we should rely on simple test. we basically have #[cfg(test)] on top which makes unhappy for clippy :).

Reply:
Gotcha. In that case, perhaps it's worth removing? I see some of the proptests include a tombstone.

Reply:
I added the test to resolve previous clippy throwing errors which resulted in an  unused  method of add_tombstone function, if I remove test case we will run into that error.

Reply:
Adding a `#[allow(dead_code)]` should fix it